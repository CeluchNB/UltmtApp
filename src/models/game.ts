import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
