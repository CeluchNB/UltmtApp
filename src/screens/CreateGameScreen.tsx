import { CreateGame } from '../types/game'
import { CreateGameProps } from '../types/navigation'
// import { GuestTeam } from '../types/team'
import LabeledFormInput from '../components/molecules/LabeledFormInput'
import PrimaryButton from '../components/atoms/PrimaryButton'
import React from 'react'
import ScreenTitle from '../components/atoms/ScreenTitle'
import { Tournament } from '../types/tournament'
// import { createGame } from '../services/data/game'
import { selectAccount } from '../store/reducers/features/account/accountReducer'
import { useColors } from '../hooks'
import { useSelector } from 'react-redux'
import { Controller, useForm } from 'react-hook-form'
import { SafeAreaView, StyleSheet, Switch, View } from 'react-native'

const CreateGameScreen: React.FC<CreateGameProps> = ({ route }) => {
    const { teamOne } = route.params
    const { colors } = useColors()
    const account = useSelector(selectAccount)

    const { control, handleSubmit } = useForm({
        defaultValues: {
            teamTwo: '',
            scoreLimit: 15,
            halfScore: 8,
            softcapMins: 75,
            hardcapMins: 90,
            playersPerPoint: 7,
            timeoutPerHalf: 1,
            floaterTimeout: true,
            tournament: {} as Tournament,
        },
    })

    const onNumberChange = (text: any): string => {
        return text.toString().replace(/\D/, '')
    }

    const onCreate = async (formData: {
        teamTwo: string
        scoreLimit: number
        halfScore: number
        softcapMins: number
        hardcapMins: number
        playersPerPoint: number
        timeoutPerHalf: number
        floaterTimeout: boolean
        tournament: Tournament
    }) => {
        const data: CreateGame = {
            ...formData,
            teamTwoDefined: false,
            startTime: new Date(),
            teamTwo: {
                name: formData.teamTwo,
            },
            teamOne: teamOne || {
                _id: 'dummy1',
                place: 'Place1',
                name: 'Name1',
                teamname: 'teamname1',
                seasonStart: '2022',
                seasonEnd: '2023',
            },
            creator: {
                _id: account.email,
                firstName: account.firstName,
                lastName: account.lastName,
                username: account.username,
            },
        }
        console.log('creating with data', data)
        // await createGame(data)
    }

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        container: {
            width: '80%',
            alignSelf: 'center',
        },
        teamItemContainer: {
            display: 'flex',
            flexDirection: 'row',
        },
    })

    return (
        <SafeAreaView style={styles.screen}>
            <View style={styles.container}>
                {teamOne ? (
                    <ScreenTitle title={teamOne.name} />
                ) : (
                    <ScreenTitle title="New Game" />
                )}
                <ScreenTitle title="vs." />
                <ScreenTitle title="Team Two" />
                <Controller
                    name="scoreLimit"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 1,
                        maxLength: 2,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Game to"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="points"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="halfScore"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 1,
                        maxLength: 2,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Half at"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="points"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="softcapMins"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 2,
                        maxLength: 3,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Soft cap"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="minutes"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="hardcapMins"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 2,
                        maxLength: 3,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Hard cap"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="points"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="playersPerPoint"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 1,
                        maxLength: 2,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Players"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="per point"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="timeoutPerHalf"
                    control={control}
                    rules={{
                        required: true,
                        minLength: 1,
                        maxLength: 2,
                        pattern: /\d+/,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Timeouts"
                                onChange={event => {
                                    onChange(onNumberChange(event))
                                }}
                                value={value}
                                unit="per half"
                                keyboardType="number-pad"
                            />
                        )
                    }}
                />
                <Controller
                    name="floaterTimeout"
                    control={control}
                    rules={{
                        required: true,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Floater timeout"
                                onChange={onChange}
                                value={value}>
                                <Switch
                                    trackColor={{
                                        false: colors.gray,
                                        true: colors.textSecondary,
                                    }}
                                    thumbColor={colors.textPrimary}
                                    onValueChange={onChange}
                                    value={value}
                                />
                            </LabeledFormInput>
                        )
                    }}
                />
                <Controller
                    name="tournament"
                    control={control}
                    rules={{
                        required: true,
                    }}
                    render={({ field: { onChange, value } }) => {
                        return (
                            <LabeledFormInput
                                label="Tournament"
                                onChange={onChange}
                                value={value}
                            />
                        )
                    }}
                />
                <PrimaryButton
                    text="start"
                    loading={false}
                    onPress={handleSubmit(onCreate)}
                />
            </View>
        </SafeAreaView>
    )
}

export default CreateGameScreen
