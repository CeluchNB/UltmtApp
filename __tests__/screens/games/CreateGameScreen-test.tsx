import * as GameData from '../../../src/services/data/game'
import { CreateGameProps } from '../../../src/types/navigation'
import CreateGameScreen from '../../../src/screens/games/CreateGameScreen'
import MockDate from 'mockdate'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { setProfile } from '../../../src/store/reducers/features/account/accountReducer'
import { setTeamOne } from '../../../src/store/reducers/features/game/liveGameReducer'
import store from '../../../src/store/store'
import { act, fireEvent, render } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const teamOne = {
    _id: 'team1',
    place: 'Place 1',
    name: 'Name 1',
    teamname: 'place1name1',
    seasonStart: '2022',
    seasonEnd: '2022',
}

const teamTwo = {
    _id: 'team2',
    place: 'Place 2',
    name: 'Name 2',
    teamname: 'place2name2',
    seasonStart: '2022',
    seasonEnd: '2022',
}

const props: CreateGameProps = {
    navigation: {} as any,
    route: { params: { teamOne, teamTwo } } as any,
}

beforeAll(() => {
    MockDate.set('2022')
    store.dispatch(
        setProfile({
            _id: 'id1',
            firstName: 'First',
            lastName: 'Last',
            email: 'email@email.com',
            username: 'username',
            error: undefined,
            leaveManagerError: undefined,
            privateProfile: false,
            openToRequests: true,
            playerTeams: [],
            managerTeams: [],
            archiveTeams: [],
            requests: [],
            toggleRosterStatusLoading: false,
            editLoading: false,
            fetchProfileLoading: false,
        }),
    )
    store.dispatch(
        setTeamOne({
            ...teamOne,
            managers: [],
            players: [],
            seasonNumber: 2,
            continuationId: 'place1name1',
            rosterOpen: true,
            requests: [],
            games: [],
        }),
    )
})

afterAll(() => {
    MockDate.reset()
})

it('should match snapshot', () => {
    const snapshot = render(
        <Provider store={store}>
            <NavigationContainer>
                <CreateGameScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    expect(snapshot.toJSON()).toMatchSnapshot()
})

it('should create game', async () => {
    const spy = jest
        .spyOn(GameData, 'createGame')
        .mockReturnValue(Promise.resolve({} as any))
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <CreateGameScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const button = getByText('start')
    fireEvent.press(button)

    // It would be better if I could wait for
    // a loading indicator to be gone
    await act(async () => {})

    expect(spy).toHaveBeenCalledWith(
        {
            teamOne,
            teamTwo,
            creator: {
                _id: 'id1',
                firstName: 'First',
                lastName: 'Last',
                username: 'username',
            },
            scoreLimit: 15,
            halfScore: 8,
            softcapMins: 75,
            hardcapMins: 90,
            playersPerPoint: 7,
            timeoutPerHalf: 1,
            floaterTimeout: true,
            teamTwoDefined: true,
            startTime: new Date(),
            tournament: undefined,
        },
        false,
        [],
    )
})
