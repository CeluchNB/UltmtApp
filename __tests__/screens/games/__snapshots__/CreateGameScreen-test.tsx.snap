// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#ffd600",
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Name 1
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#ffd600",
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        vs.
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#ffd600",
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Name 2
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Game to
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Game to"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="15"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        points
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Half at
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Half at"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="8"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        points
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Soft cap
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Soft cap"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="75"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        minutes
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Hard cap
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Hard cap"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="90"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        points
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Players
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Players"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="7"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        per point
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Timeouts
      </Text>
      <View
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <View
          style={
            Object {
              "placeholderTextColor": "#ffffff",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#303030",
                  "borderColor": "#31adff",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                editable={true}
                keyboardType="number-pad"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Timeouts"
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value="1"
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ababab",
            "fontSize": 30.821917808219176,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        per half
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffd600",
            "fontSize": 41.0958904109589,
            "fontWeight": "bold",
            "width": "50%",
          }
        }
      >
        Floater timeout
      </Text>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#31adff"
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#ffd600"
          tintColor="#ababab"
          value={true}
        />
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffd600",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "minWidth": "auto",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  Object {
                    "marginHorizontal": 8,
                  },
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            start
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
